src/test/java/com/commercetools/sync/shoppinglists/utils/ShoppingListsReferenceResolutionUtilsTest.java

```diff
@@ -0,0 +1,193 @@
+        final Customer mockCustomer = getMockCustomer(UUID.randomUUID().toString(), "customerKey");
+        final ProductVariant mockProductVariant = mock(ProductVariant.class);
+        when(mockProductVariant.getSku()).thenReturn("variant-sku");
+
+        final List<ShoppingList> mockShoppingLists = new ArrayList<>();
+        mockShoppingLists.add(null);
+
+        for (int i = 0; i < 3; i++) {
+            final ShoppingList mockShoppingList = mock(ShoppingList.class);
+
+            final Reference<Customer> customerReference =
+                Reference.ofResourceTypeIdAndObj(Customer.referenceTypeId(), mockCustomer);
+            when(mockShoppingList.getCustomer()).thenReturn(customerReference);
+
+            final CustomFields mockCustomFields = mock(CustomFields.class);
+            final Reference<Type> typeReference =
+                Reference.ofResourceTypeIdAndObj(Type.referenceTypeId(), mockCustomType);
+
+            when(mockCustomFields.getType()).thenReturn(typeReference);
+            when(mockShoppingList.getCustom()).thenReturn(mockCustomFields);
+
+            final LineItem mockLineItem = mock(LineItem.class);
+            when(mockLineItem.getVariant()).thenReturn(mockProductVariant);
+            when(mockLineItem.getCustom()).thenReturn(mockCustomFields);
+
+            when(mockShoppingList.getLineItems())
+                .thenReturn(singletonList(mockLineItem));
+
+            final TextLineItem mockTextLineItem = mock(TextLineItem.class);
+            when(mockTextLineItem.getName()).thenReturn(LocalizedString.ofEnglish("textLineItemName"));
+            when(mockTextLineItem.getCustom()).thenReturn(mockCustomFields);
+
+            when(mockShoppingList.getTextLineItems()).thenReturn(singletonList(mockTextLineItem));
+
+            mockShoppingLists.add(mockShoppingList);
+        }
+
+        final List<ShoppingListDraft> shoppingListDrafts = mapToShoppingListDrafts(mockShoppingLists);
+
+        assertThat(shoppingListDrafts).hasSize(3);
+        shoppingListDrafts.forEach(draft -> {
+            assertThat(draft.getCustomer().getKey()).isEqualTo("customerKey");
+            assertThat(draft.getCustom().getType().getKey()).isEqualTo("customTypeKey");
+
+            assertThat(draft.getLineItems()).containsExactly(
+                LineItemDraftBuilder
+                    .ofSku("variant-sku", 0L)
+                    .custom(CustomFieldsDraft.ofTypeKeyAndJson("customTypeKey", emptyMap()))
+                    .build());
+
+            assertThat(draft.getTextLineItems()).containsExactly(
+                TextLineItemDraftBuilder
+                    .of(LocalizedString.ofEnglish("textLineItemName"), 0L)
+                    .custom(CustomFieldsDraft.ofTypeKeyAndJson("customTypeKey", emptyMap()))
+                    .build());
+        });
+    }
+
+    @Test
+    void mapToShoppingListDrafts_WithNonExpandedReferences_ShouldReturnResourceIdentifiersWithKeys() {
```

Comment:
If you don't expand, how can it know the keys?

Reply:
You are right, renamed the test name: https://github.com/commercetools/commercetools-sync-java/pull/602/commits/2fdb1c3a8a1d24a53fa441040e53fd85c841c0d6